libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1344,34 +1344,43 @@ bool AP_AHRS_NavEKF::healthy(void) const
     return AP_AHRS_DCM::healthy();
 }
 
-bool AP_AHRS_NavEKF::prearm_healthy(void) const
+// returns false if we fail arming checks, in which case the buffer will be populated with a failure message
+bool AP_AHRS_NavEKF::pre_arm_check(char *failure_msg, uint8_t failure_msg_len) const
 {
-    bool prearm_health = false;
+    const char *ahrs_prefix = "AHRS: ";
     switch (ekf_type()) {
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     case EKFType::SITL:
 #endif
     case EKFType::NONE:
-        prearm_health = true;
-        break;
+        if (!healthy()) {
+            hal.util->snprintf(failure_msg, failure_msg_len, "AHRS not healthy");
```

Comment:
```suggestion
            hal.util->snprintf(failure_msg, failure_msg_len, "Not healthy");
```

Reply:
I've included all the changes suggested by @peterbarker